### PR TITLE
spelling corrections, some type corrections & doc.

### PR DESCRIPTION
**Pull request by rocky**
Pull out get_section_tests as its own method.

### CODE REVIEW

File: mathics/doc/common_doc.py [hunk 1]
```diff
@@ -679,6 +675,32 @@ def get_section(self, part_slug, chapter_slug, section_slug):
                 return chapter.sections_by_slug.get(section_slug)
         return None
 
+    def get_section_tests(self, section):
```
Comment (mmatera) [inline mathics/doc/common_doc.py:678]: Whereis this called?

Comment (rocky) [reply mathics/doc/common_doc.py:678]: Ah good. This has been fixed now and you'll see the code deleted from below. 

This was part of the abortive try to get more test results from Mathics3 modules into the LaTeX and Django result output.

**Comment by mmatera** [comment]
For some reason, after merging this, ~ 2/3 of the doctests are not loaded.

**Comment by rocky** [comment]
I will unmerge this.

**Comment by mmatera** [comment]
Don't do that. Just restore "mathics/doc/common_doc.py"

**Pull request merged**